numpy/polynomial/_polybase.py

```diff
@@ -283,10 +315,78 @@ def __repr__(self):
         name = self.__class__.__name__
         return f"{name}({coef}, domain={domain}, window={window})"
 
+    def __format__(self, fmt_str):
```

Comment:
Should the argument really be *required*?  Why not default to the `__str__` behavior?  As it is, for a polynomial `p` such as `p = np.polynomial.Polynomial([1, 2, 3])`, both `format(p)` and `f'{p}'` raise an exception. E.g.

```
In [138]: p = np.polynomial.Polynomial([1, 2, 3])

In [139]: f'{p}'
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-139-8e0bd8161cf8> in <module>
----> 1 f'{p}'

~/mc37np/lib/python3.7/site-packages/numpy-1.19.0.dev0+56d5197-py3.7-macosx-10.9-x86_64.egg/numpy/polynomial/_polybase.py in __format__(self, fmt_str)
    319         if fmt_str not in ('ascii', 'unicode'):
    320             raise TypeError(
--> 321                 f"Unsupported format string '{fmt_str}' passed to "
    322                 f"{self.__class__}.__format__. Valid options are "
    323                 f"'ascii' and 'unicode'"

TypeError: Unsupported format string '' passed to <class 'numpy.polynomial.polynomial.Polynomial'>.__format__. Valid options are 'ascii' and 'unicode'
```

Reply:
Or rather, when the argument is `''`, use the same logic as `__str__`.